src/Model/Config/ApplicationConfig.php

```
@@ -0,0 +1,43 @@
+<?php
+
+namespace Supermarket\Model\Config;
+
+use Supermarket\Model\Config\ApplicationConfig\DatabaseConfig;
+use Supermarket\Model\Config\ApplicationConfig\TemplateConfig;
+
+class ApplicationConfig
+{
+    /**
+     * @var DatabaseConfig
+     */
+    private $databaseCredentials;
+
+    /**
+     * @var TemplateConfig
+     */
+    private $templateConfig;
+
+    public static function createFromArray(array $data): ApplicationConfig
+    {
+        return new self(
+            DatabaseConfig::createFromArray($data['db']['connection']),
```

use private consts for keys, also validation missing (need to throw invalid argument exception in case the key is not defined